Add wlan interface to domd

dsemenets opened the pull request.

lorc commented on recipes-kernel/linux/files/bcm2712-raspberrypi5-domd.dts:
```diff
@@ -34,13 +34,72 @@
 			sdhci-caps-mask = <0xc000 0x0>;
 			xen,path = "/axi/mmc@fff000";
-			xen,force-assign-without-iommu = <0x1>;
+			xen,force-assign-without-iommu;
```
This is the correct change, but it should be in the separate commit, as it merely fixes prior mistake.

dsemenets replied:
Moved to separate PR

GrygiriiS commented:
Sorry, but It is not how it should be - we can't blindly and unconditionally add devices in Dom1 (DomD).
For example, now we pass MMC and Sd-card is used for rootfs, but our target is USB and USB stick with rootfs.
Networking, 4 cases possible: no net; Ethernet; wifi;  Ethernet and wifi. And it should be configurable. 

So there should be build options and DT split somehow (probably overlays). So final DT and rootfs are built based on configuration options..

lorc commented:
NAK. Please read https://xen-troops.github.io/process/process

`linux: domd.dts: Minor fixes.` is not a valid commit description

oleksiimoisieiev commented:
@dsemenets Are you going to fix comments from colleagues?

dsemenets committed:
linux: Modify device tree for use wireless in domd

Passthrough nodes in xen dtb:
* gio
* sdio2
* pinctrl
Add dt overlay with appropriate nodes to domd's device tree.

Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
Reviewed-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>

dsemenets committed:
xt-rpi-u-boot-scr: Load dt overlays to wifi support in domd

Load appropriate DT overlays to wifi support in domd

Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
Reviewed-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>

oleksiimoisieiev commented on recipes-kernel/linux/files/bcm2712-raspberrypi5-wifi.dtso:
```diff
@@ -0,0 +1,127 @@
+/*
```
gio? Maybe gpio

dsemenets replied:
No. gpio7d50800 is gio in rpi5 dt

GrygiriiS replied:
Yep. it's gio

oleksiimoisieiev commented on recipes-kernel/linux/files/bcm2712-raspberrypi5-wifi.dtso:
```diff
@@ -0,0 +1,127 @@
+		};
+	};
+
+	mmc@1001100000 {
```
do we need mmc for wifi?

dsemenets replied:
wifi is child for this node

oleksiimoisieiev commented on recipes-bsp/xt-rpi-u-boot-scr/xt-rpi-u-boot-scr.bb:
```diff
@@ -16,6 +16,7 @@ SRC_URI = " \
 "
 MMC_PASSTHROUGH_DTBO = "mmc-passthrough.dtbo"
 USB_PASSTHROUGH_DTBO = "usb-passthrough.dtbo"
+WIFI_PASSTHROUGH_DTBO = "wifi-passthrough.dtbo"
```
I don't think we need a separate dtbo. It should be configurable

dsemenets replied:
This dtbo applied if MACHINE_FEATURES contains domd_wifi

oleksiimoisieiev replied:
I understand. but I see here that we have two configurations:
domd via usb, domd via mmc and domd via wifi but usb and mmc disabled. Is it true?
I see wifi as a optional feature which either can be enabled or disabled. Or if it's mandatory or it is a problem to make it optional then don't add this option

dsemenets replied:
It is optional feature. It can be used with rootfs on usb or mmc. Overlays are not applied if MACHINE_FEATURES doesn't contain domd_wifi

oleksiimoisieiev replied:
So why do we have 3 dtbo ? for mmc, usb and wifi?

GrygiriiS replied:
The overlays applied to add features: 
```
usb-passthrough.dtbo = USB + NET
usb-passthrough.dtbo + wifi-passthrough.dtbo = USB + NET + WIFI
mmc-passthrough.dtbo + wifi-passthrough.dtbo = MMC + NET + WIFI
mmc-passthrough.dtbo + wifi-passthrough.dtbo + usb-passthrough.dtbo = MMC + USB +NET + WIFI

... etc
```

dsemenets replied:
Yes. But combination mmc-passthrough.dtbo + wifi-passthrough.dtbo + usb-passthrough.dtbo wasn't tested

GrygiriiS replied:
Of course not, it was a generic answer on generic question "Why we need  dtbo"

dsemenets replied:
> So why do we have 3 dtbo ? for mmc, usb and wifi?

Yes

oleksiimoisieiev replied:
Ok thanks

GrygiriiS commented on recipes-kernel/linux/files/bcm2712-raspberrypi5-wifi.dtso:
```diff
@@ -0,0 +1,127 @@
+/plugin/;
+
+&passthrough {
+	pinctrl@107d504100 {
```
pinctrl looks like a problem, unfortunately :( We can't jsut passthrough it, as it's shared resource. All current HW working with assumption pinctrl is configured already by RPI5 bootloader or u-boot.

GrygiriiS replied:
UPD: Not sure how to proceed :(

lorc replied:
Yeah, in cases like this we are considering using SCMI. This is the best approach, but it is PITA to implement. 

Another approach is to configure pinctrl in bootloader/dom0 so guest can use static configuration. Of course, it will not work if dynamic configuration is required.

oleksiimoisieiev commented:
LGTM `Reviewed-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>`

lorc commented:
We are passing SoC pinctrl to DomD, which is not secure, as DomD may (and probably will) mess with SDHC used in Dom0.

Therefore, it is NAK for current implementation. Please consider using SCMI or configuring pinctrl from a bootloader.

oleksiimoisieiev commented:
> We are passing SoC pinctrl to DomD, which is not secure, as DomD may (and probably will) mess with SDHC used in Dom0.
> 
> Therefore, it is NAK for current implementation. Please consider using SCMI or configuring pinctrl from a bootloader.

I understand your concerns and agree with them. Looks like making pinctrl config on u-boot is the only option for us. That's because we don't have a full support for arm-trusted-firmware. This means that SCMI support implementation may require a lot of effort (unless I've missed something and rpi5 loader has scmi support).
So the only option I see for now is to make u-boot support. Especially counting the fact that u-boot already has pinctl driver.
@GrygiriiS @lorc what do you think?

lorc commented:
> (unless I've missed something and rpi5 loader has scmi support).

No, but we have TF-A running, so we can use TF-A based SCMI. I believe, you are the author of these patches?

> So the only option I see for now is to make u-boot support. 

It makes the whole system depending on U-Boot which is already bad. But bigger problem is that we can't change pinctrl (and clocks and resets) dynamically.

dsemenets committed:
xt-rpi-u-boot-scr: Rename variables in recipe

Rename XEN_DTBO_ADDR variable to more appropriate DTBO_LOAD_ADDR

Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>

klogg commented:
@oleksiimoisieiev will take ownership of these patches

oleksiimoisieiev commented:
> @oleksiimoisieiev will take ownership of these patches

Taking to work. waiting for SCMI to be finished.

GrygiriiS commented:
Not needed any more. all required parts incorporated in
https://github.com/xen-troops/meta-xt-prod-devel-rpi5/pull/41
